seaborn/axisgrid.py

```diff
@@ -1797,6 +1797,47 @@ def plot_marginals(self, func, **kwargs):
 
         return self
 
+    def refline(self, *, x=None, y=None, joint=True, marginal=True, **line_kws):
+        """Add a reference line to joint and/or marginal axes.
+
+        Parameters
+        ----------
+        x, y : numeric
+            Value to draw the line at.
+        joint, marginal : bools
+            Whether to add the reference line to the joint/marginal axes.
+        line_kws : key, value mappings
+            Other keyword arguments are passed to :meth:`matplotlib.axes.Axes.axvline`
+            when ``x`` is not None or :meth:`matplotlib.axes.Axes.axhline` when ``y``
+            is not None.
+
+        Returns
+        -------
+        :class:`JointGrid` instance
+            Returns ``self`` for easy method chaining.
+
+        """
+        if x is None and y is None:
+            raise ValueError('Must provide ``x`` or ``y``')
+        elif x is not None and y is not None:
+            raise ValueError('Provide either ``x`` or ``y``, not both')
```

Comment:
Why have this limitation?

Reply:
I was aligning this with the `axhline()` / `axvline()` behavior i.e. 1 call for each line you want. How did you initially envision this?

Reply:
I don't see any reason not to allow `g.refline(x=..., y=...)`.

Reply:
I'll make an update then.